crates/rover-client/src/shared/check_response.rs

```diff
@@ -10,42 +10,80 @@ use serde::{Deserialize, Serialize};
 use prettytable::{row, Table};
 use serde_json::{json, Value};
 
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub enum CheckResponse {
+    OpeartionCheckResponse(OperationCheckResponse),
+    OpeartionLessCheckResponse(OpeartionLessCheckResponse),
+}
+
+impl CheckResponse {
+    pub fn get_json(&self) -> Value {
+        match self {
+            CheckResponse::OpeartionCheckResponse(operation_check_response) => {
+                operation_check_response.get_json()
+            }
+            CheckResponse::OpeartionLessCheckResponse(operation_less_check_response) => {
+                operation_less_check_response.get_json()
+            }
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub struct OpeartionLessCheckResponse {
+    pub target_url: Option<String>,
+    pub core_schema_modified: bool,
+    pub contains_operation_check: bool,
+}
+
+impl OpeartionLessCheckResponse {
+    pub fn to_output(&self) -> String {
+        "".to_string()
+    }
+
+    pub fn get_json(&self) -> Value {
+        json!(self)
+    }
+}
+
 /// CheckResponse is the return type of the
 /// `graph` and `subgraph` check operations
 #[derive(Debug, Serialize, Clone, Eq, PartialEq)]
-pub struct CheckResponse {
+pub struct OperationCheckResponse {
     target_url: Option<String>,
     operation_check_count: u64,
     changes: Vec<SchemaChange>,
     #[serde(skip_serializing)]
     result: ChangeSeverity,
     failure_count: u64,
     core_schema_modified: bool,
+    contains_operation_check: bool,
 }
 
-impl CheckResponse {
+impl OperationCheckResponse {
     pub fn try_new(
         target_url: Option<String>,
         operation_check_count: u64,
         changes: Vec<SchemaChange>,
         result: ChangeSeverity,
         graph_ref: GraphRef,
         core_schema_modified: bool,
-    ) -> Result<CheckResponse, RoverClientError> {
+    ) -> Result<OperationCheckResponse, RoverClientError> {
         let mut failure_count = 0;
         for change in &changes {
             if let ChangeSeverity::FAIL = change.severity {
                 failure_count += 1;
             }
         }
 
-        let check_response = CheckResponse {
+        let check_response = OperationCheckResponse {
             target_url,
             operation_check_count,
             changes,
             result,
             failure_count,
             core_schema_modified,
+            contains_operation_check: true,
```

Comment:
I think we don't need this since `OperationCheckResponse` is nested in the enum, so we know this is an operation check response.